File: app/src/main/java/de/tum/in/tumcampusapp/models/tumo/Geo.java
package de.tum.in.tumcampusapp.models.tumo;

import org.simpleframework.xml.Element;
import org.simpleframework.xml.Root;

@Root(name = "geo")
public class Geo {
    @Element(required = false)
    private String latitude = "0";

    @Element(required = false)
    private String longitude = "0";

    public Geo() {
    }

    public Geo(double latitude, double longitude) {
        this.latitude = Double.toString(latitude);
        this.longitude = Double.toString(longitude);
    }

    public String getLatitude() {
        return latitude;
    }

    public String getLongitude() {
        return longitude;
    }

    @Override
    public String toString() {
        return "latitude=" + latitude + " longitude=" + longitude;
    }
}

File: app/src/main/java/de/tum/in/tumcampusapp/models/tumo/Geo.kt
package de.tum.`in`.tumcampusapp.models.tumo

import org.simpleframework.xml.Element
import org.simpleframework.xml.Root

@Root(name = "geo")
data class Geo(@field:Element(required = false) var latitude: String = "0",
               @field:Element(required = false) var longitude: String = "0") {

    constructor(latitude: Double, longitude: Double) : this(
            latitude = java.lang.Double.toString(latitude),
            longitude = java.lang.Double.toString(longitude)
    )

    override fun toString(): String = "latitude=$latitude longitude=$longitude"
}